packages/react-router/src/router.ts

```diff
@@ -1816,12 +1816,16 @@ export class Router<
     // Reset the view transition flag
     delete this.shouldViewTransition
     // Attempt to start a view transition (or just apply the changes if we can't)
-    ;(shouldViewTransition && typeof document !== 'undefined'
-      ? document
-      : undefined
-    )
-      // @ts-expect-error
-      ?.startViewTransition?.(fn) || fn()
+    if (
+      shouldViewTransition &&
```

Comment:
Unfortunately had to rework the code a bit here to pass both 5.6 and the rest :(